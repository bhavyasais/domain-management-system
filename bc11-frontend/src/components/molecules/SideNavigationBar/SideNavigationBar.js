import React, { useState } from 'react';
import { withStyles } from '@material-ui/core/styles';
import { Grid, Typography } from '@material-ui/core';
import sideNavIconGetter from '../../../utils/sideNavIconGetter';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import { useHistory } from 'react-router-dom';

const styles = (theme) => ({
  root: {
    borderRight: '1px solid #eeeeee',
    height: '100vh',
    width: '64px',
  },
  image: {
    heigth: '24px',
    width: '24px',
    marginBottom: '22px',
    marginTop: '22px',
    cursor: 'pointer',
  },
  imageContainer: {
    width: '64px',
    justifyContent: 'center',
  },
  activeIcon: {
    backgroundColor: '#eaf0ff',
    borderLeft: '2px solid #1d4cd7',
  },
  header: {
    textAlign: 'center',
    alignSelf: 'center',
    color: '#1d4cd7',
    fontSize: '28px',
    fontWeight: '300',
    marginBottom: '18px',
    marginTop: '18px',
    width: '64px',
    wordWrap: 'break-word',
  },
});

const SideNavigationBar = ({ classes, header }) => {
  const iconList = sideNavIconGetter.getSideBarIcons();
  const [currentIconIndex, setCurrentIconIndex] = useState(1);
  const history = useHistory();
  const handleOnClick = (index, active) => () => {
    if (active) {
      setCurrentIconIndex(index);
      switch (index) {
        case 1:
          history.push('/base');
          break;
        case 2:
          history.push('/emailIntegration');
          break;
      }
    }
  };

  return (
    <Grid container direction="column" className={classes.root}>
      <Grid item>
        <Typography className={classes.header}>{header}</Typography>
      </Grid>

      {iconList.map((item, index) => (
        <Grid
          item
          container
          className={clsx(classes.imageContainer, currentIconIndex === index && classes.activeIcon)}
          key={index}
        >
          <img
            src={item.icon}
            className={classes.image}
            onClick={handleOnClick(index, item.active)}
          />
        </Grid>
      ))}
    </Grid>
  );
};

SideNavigationBar.propTypes = {
  header: PropTypes.string,
  onClick: PropTypes.func,
};
SideNavigationBar.defaultProps = {
  onClick: null,
  header: 'io',
};

export default withStyles(styles, { withTheme: true })(SideNavigationBar);
