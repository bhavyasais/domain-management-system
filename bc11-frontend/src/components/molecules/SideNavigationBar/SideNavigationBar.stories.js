import React from "react";
import { withKnobs, text } from "@storybook/addon-knobs";
import SideNavigationBar from "./SideNavigationBar";

export default {
  component: SideNavigationBar,
  title: "Molecules/SideNavigationBar",
  decorators: [withKnobs],
};

export const defaultSideNavBar = () => (
  <SideNavigationBar header={text("header", "io")} />
);
