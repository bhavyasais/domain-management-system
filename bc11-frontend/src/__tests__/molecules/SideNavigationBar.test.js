import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import SideNavigationBar from '../../components/molecules/SideNavigationBar/SideNavigationBar';

describe('<SideNavigationBar/>', () => {
  const history = { push: jest.fn() };
  let mounted;
  let onClick;
  beforeEach(() => {
    onClick = jest.fn();
    act(() => {
      mounted = mount(<SideNavigationBar onClick={onClick} history={history} />);
    });
  });

  it('Renders correctly', () => {
    let wrapper;
    wrapper = renderer.create(
      <SideNavigationBar header="io" onClick={onClick} history={history} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("onClick does'nt works on clicking inactive icon", () => {
    mounted.find('img').at(0).props().onClick();
    expect(onClick).not.toHaveBeenCalled();
  });
});
