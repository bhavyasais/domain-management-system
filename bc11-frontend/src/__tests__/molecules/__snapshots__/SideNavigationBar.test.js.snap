// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNavigationBar/> Renders correctly 1`] = `
<div
  className="MuiGrid-root SideNavigationBar-root-1 MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <p
      className="MuiTypography-root SideNavigationBar-header-5 MuiTypography-body1"
    >
      io
    </p>
  </div>
  <div
    className="MuiGrid-root SideNavigationBar-imageContainer-3 MuiGrid-container MuiGrid-item"
  >
    <img
      className="SideNavigationBar-image-2"
      onClick={[Function]}
      src="atoms_icon_pie.svg"
    />
  </div>
  <div
    className="MuiGrid-root SideNavigationBar-imageContainer-3 SideNavigationBar-activeIcon-4 MuiGrid-container MuiGrid-item"
  >
    <img
      className="SideNavigationBar-image-2"
      onClick={[Function]}
      src="atoms_icon_graph.svg"
    />
  </div>
  <div
    className="MuiGrid-root SideNavigationBar-imageContainer-3 MuiGrid-container MuiGrid-item"
  >
    <img
      className="SideNavigationBar-image-2"
      onClick={[Function]}
      src="mail.png"
    />
  </div>
  <div
    className="MuiGrid-root SideNavigationBar-imageContainer-3 MuiGrid-container MuiGrid-item"
  >
    <img
      className="SideNavigationBar-image-2"
      onClick={[Function]}
      src="atoms_icon_excel.svg"
    />
  </div>
</div>
`;
